ios/RNSScreenStackHeaderConfigComponentView.mm

```diff
@@ -0,0 +1,679 @@
+  // It is workaround for loading custom back icon when transitioning from a screen without header to the screen which
+  // has one. This action fails when navigating to the screen with header for the second time and loads default back
+  // button. It looks like changing the tint color of navbar triggers an update of the items belonging to it and it
+  // seems to load the custom back image so we change the tint color's alpha by a very small amount and then set it to
+  // the one it should have.
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && defined(__IPHONE_14_0) && \
+    __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_14_0
+  // it brakes the behavior of `headerRight` in iOS 14, where the bug desribed above seems to be fixed, so we do nothing
+  // in iOS 14
+  if (@available(iOS 14.0, *)) {
+  } else
+#endif
+  {
+    [navbar setTintColor:[config.color colorWithAlphaComponent:CGColorGetAlpha(config.color.CGColor) - 0.01]];
+  }
+  [navbar setTintColor:config.color];
+
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && defined(__IPHONE_13_0) && \
+    __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_13_0
+  if (@available(iOS 13.0, *)) {
+    // font customized on the navigation item level, so nothing to do here
+  } else
+#endif
+  {
+    BOOL hideShadow = config.hideShadow;
+
+    if (config.backgroundColor && CGColorGetAlpha(config.backgroundColor.CGColor) == 0.) {
+      [navbar setBackgroundImage:[UIImage new] forBarMetrics:UIBarMetricsDefault];
+      [navbar setBarTintColor:[UIColor clearColor]];
+      hideShadow = YES;
+    } else {
+      [navbar setBackgroundImage:nil forBarMetrics:UIBarMetricsDefault];
+      [navbar setBarTintColor:config.backgroundColor];
+    }
+    [navbar setTranslucent:config.translucent];
+    [navbar setValue:@(hideShadow ? YES : NO) forKey:@"hidesShadow"];
+
+    if (config.titleFontFamily || config.titleFontSize || config.titleFontWeight || config.titleColor) {
+      NSMutableDictionary *attrs = [NSMutableDictionary new];
+
+      if (config.titleColor) {
+        attrs[NSForegroundColorAttributeName] = config.titleColor;
+      }
+
+      NSString *family = config.titleFontFamily ?: nil;
+      NSNumber *size = config.titleFontSize ?: @17;
+      NSString *weight = config.titleFontWeight ?: nil;
+      if (family || weight) {
+        attrs[NSFontAttributeName] = [RCTFont updateFont:nil
+                                              withFamily:family
+                                                    size:size
+                                                  weight:weight
+                                                   style:nil
+                                                 variant:nil
+                                         scaleMultiplier:1.0];
+      } else {
+        attrs[NSFontAttributeName] = [UIFont boldSystemFontOfSize:[size floatValue]];
+      }
+      [navbar setTitleTextAttributes:attrs];
+    }
+
+#if !TARGET_OS_TV
+    if (@available(iOS 11.0, *)) {
```

Comment:
I think it's okay to remove iOS 11 checks hence react-native doesn't support iOS 10 anymore.